### PR TITLE
Levenshtein algorithm for string-like types

### PR DESCRIPTION
**Pull request by snim2**
This PR is needed to implement the Zhang-Shasha algorithm, it adds:

  * Levenshtein distance for string-like types
  * Changes LCSS algorithm to use fixed capacity vectors.

No user-facing changes. 

**Needs squashing.**

### CODE REVIEW

File: src/lib/sequence.rs [hunk 1]
```diff
@@ -88,6 +92,51 @@ pub fn lcss<T: Clone + Eq>(seq1: &[NodeId],
     lcss
 }
 
+/// Three-way minimum.
+fn _min3<T: Ord>(v1: T, v2: T, v3: T) -> T {
```
Comment (ltratt) [inline src/lib/sequence.rs]: Why `_min3` with an underscore?

File: src/lib/sequence.rs [hunk 2]
```diff
@@ -88,6 +92,51 @@ pub fn lcss<T: Clone + Eq>(seq1: &[NodeId],
...
+}
+
+/// Levenshtein distance with equal weights for deletion, insertion and substitution.
+pub fn levenshtein<T: Eq + PartialEq + ToString>(s1: &T, s2: &T) -> usize {
+    if s1 == s2 {
+        return 0;
+    }
+    let w1 = s1.to_string().chars().collect::<Vec<_>>();
+    let w2 = s2.to_string().chars().collect::<Vec<_>>();
+    if w1.is_empty() {
```
Comment (ltratt) [inline src/lib/sequence.rs]: It would be quicker to move these checks to before the `collect`.

**Comment by snim2** [comment]
Ready for squashing?

**Comment by ltratt** [comment]
Yep.

**Comment by snim2** [comment]
Splat.

**Pull request merged**